sled-agent/src/config.rs

```diff
@@ -53,4 +57,13 @@ impl Config {
         let config = toml::from_str(&contents)?;
         Ok(config)
     }
+
+    pub fn get_link(&self) -> Result<PhysicalLink, dladm::Error> {
```

Comment:
This is all pretty fluid, but the current proposal in RFD 259 is not to base the bootstrap address on something like a MAC, either of the sled or the SP. Instead, we're proposing either fully random at startup time, or deriving it from a separate, factory-configured value in EEPROM.

At this point, I might suggest making the interface here one that just gets that 48-bit value, and tacks it on to `fdb0:` to make the bootstrap prefix.

Reply:
Yeah, that's a good point I hadn't considered. I was hoping we could make the interface at the level of the 48-bit identifier or at least the address derived from it, rather than specifying _which_ piece of hardware the bits were taken from. Thanks for filing the issue, we'll just tackle it later when the source becomes more clear.

Reply:
Are we at the point where we're advertising some kind of serial number between sled agents, whether that's being simulated, borrowed from SMBIOS or something else?

Reply:
kiiiinda - part of the Sled Agent config is a [UUID](https://github.com/oxidecomputer/omicron/blob/2ff53f24a150311f61011e3aaf28bb583a736f13/sled-agent/src/config.rs#L21), but it's something that needs to be manually tweaked when setting up each sled (which sucks, so I don't really want to add more dependencies on it).

I really want to use a unique identifier that the sleds can generate by themselves - I figured long-term this would be the [DeviceID](https://rfd.shared.oxide.computer/rfd/0238#_a_brief_note_about_sled_identity) which is generated from the root-of-trust.

In this current implementation, that bootstrap address is acting as the "key" to identify sleds, but this could pretty easily shift to something else, if there was something else that's durable we could attach.

Reply:
I mostly asked about this because every Gimlet has a serial number which is in a read-only EEPROM, which is the main thing I was suggesting to use for correlating identity between different channels, with using the attestations and related device cert info as intermediaries there. That is, my current understanding is when we're reporting information about the sled back upstream to Oxide, it is based on its serial number. That's not to say that the intermediate device ID doesn't play an important role, but also something for us to figure out how we want to do that and a thing to think about is if the RoT changes in a future gen, what does it mean if that actual chain of trust changes (e.g. we don't use the LPC55 on say a next gen sled).

It sounds like it's more work today, though I expect it's actually something we'll be using quite heavily throughout the stack on our actual hardware.